File: src/main/java/com/bookstore/repository/AuthorRepository.java
package com.bookstore.repository;

import com.bookstore.model.Author;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface AuthorRepository extends JpaRepository<Author, UUID> {
    // Find authors by name (case-insensitive, partial match)
    List<Author> findByNameContainingIgnoreCase(String name);
} 

File: src/main/kotlin/com/bookstore/repository/AuthorRepository.kt
package com.bookstore.repository

import com.bookstore.model.Author
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface AuthorRepository : JpaRepository<Author, UUID> {
    fun findByNameContainingIgnoreCase(name: String): List<Author>

    // Paginated version of findByNameContainingIgnoreCase
    fun findByNameContainingIgnoreCase(name: String, pageable: Pageable): Page<Author>

    // Paginated version of findAll (already provided by PagingAndSortingRepository)
    // override for clarity
    override fun findAll(pageable: Pageable): Page<Author>
}
